### PR TITLE
[ready] Prevents trying to use explosive lances as an ingredient in crafting explosive lances

### PR DESCRIPTION
**Pull request by MrDoomBringer**
When making an e-lance, the recipe would try to use e-lances as an ingredient, only failing after the do_after() proc. This makes it so explosive lances aren't even considered as a possible ingredient for the recipe.

:cl: MrDoomBringer
fix: You will no longer attempt to use an explosive lance to craft an explosive lance
/:cl:

This PR also lays a foundation for blacklisting other subtypes in crafting recipes.

### CODE REVIEW

**Comment by Buggy123** [comment]
For realism, we should make it so that using a explosive lance to craft a explosive lance requires no other ingredients.

"You take your _explosive lance_, and add _nothing_, and craft a _explosive lance_. Congratulations!"

**Comment by PKPenguin321** [comment]
we should just remove explosive lances, they were added for a silly mad max joke that was funny once but are now used exclusively by powergamers to oneshot wizarda

**Comment by k-k-karma** [comment]
headmins have said in the past they're needed iirc

File: code/modules/crafting/craft.dm [hunk 1]
```diff
@@ -54,7 +54,9 @@
 /datum/personal_crafting/proc/check_contents(datum/crafting_recipe/R, list/contents)
 	contents = contents["other"]
 	main_loop:
-		for(var/A in R.reqs)
+		for(var/atom/A in R.reqs)
+			if (A.isInCraftingBlacklist(R))
```
Comment (AnturK) [inline code/modules/crafting/craft.dm]: Uh, shouldn't this be checking the actual item, not requirement ?

**Review by AnturK — CHANGES_REQUESTED** [review]
Not sure about this format, this is property of the recipe not the spear.

File: code/modules/crafting/craft.dm (hunk 1 above)
Comment (MrDoomBringer) [reply code/modules/crafting/craft.dm]: hm
perhaps i am braindead

**Comment by ExcessiveUseOfCobblestone** [comment]
headmins have no more say about balance than anyone else who plays.

File: code/modules/crafting/craft.dm (hunk 1 above)
Comment (MrDoomBringer) [reply code/modules/crafting/craft.dm]: this is gonna need a bit of work

**Comment by PKPenguin321** [comment]
@k-k-karma im sorry what?

**Comment by optimumtact** [comment]
I think he got lost on the way to hippie station

**Comment by subject217** [comment]
>This PR also lays a foundation for blacklisting other subtypes in crafting recipes.

thank u so much bb

File: code/modules/crafting/craft.dm (hunk 1 above)
Comment (MrDoomBringer) [reply code/modules/crafting/craft.dm]: @ExcessiveUseOfCobblestone this code is much harder to deal with than that, im gonna need to do some different shit

Comment (ExcessiveUseOfCobblestone) [reply code/modules/crafting/craft.dm]: I'll give it the cobblestone pass (tm) this evening.

**Comment by GuyonBroadway** [comment]
As a representative from hippie I can say that nobody there likes e-lances either. 

The guy who uses them most only does so to show how bullshit easy they are to make/win the game with.

File: code/modules/crafting/craft.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/crafting/craft.dm]: i was playing heroes of the storm whoops

**Comment by Vidiveni** [comment]
Instead of removing explosive lances, you could make them explode on the adjacent tile instead of right on top of the target. That way 100/100 potassium/water won't gib.

**Comment by subject217** [comment]
if elances are removed we can stop using them as a balance justification for other bullshit, eat your heart out kor phaeron

File: _maps/_basemap.dm [hunk 2]
```diff
@@ -1,4 +1,4 @@
-//#define LOWMEMORYMODE //uncomment this to load centcom and runtime station and thats it.
+#define LOWMEMORYMODE //uncomment this to load centcom and runtime station and thats it.
```
Comment (ShizCalev) [inline _maps/_basemap.dm]: uh

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: _maps/_basemap.dm (hunk 2 above)
Comment (optimumtact) [reply _maps/_basemap.dm]: dab

**Pull request merged**